src/pages/[platform]/build-a-backend/auth/concepts/user-attributes/index.mdx

```diff
@@ -53,16 +59,61 @@ export const auth = defineAuth({
 })
 ```
 
+## Standard attributes
+
 User attributes are defined as [Cognito Standard Attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#cognito-user-pools-standard-attributes). Attributes can be configured to be _required_ for user sign-up in addition to whether the values are _mutable_. When configuring your resource to allow your users to login with `email`, an email must be specified for user sign-up and cannot be changed later. However additional attributes can be configured to be optional, and mutable after sign-up.
 
-<Callout warning>
+## Custom attributes
 
-**Warning**: After you create your auth resource, you cannot switch an attribute between required and not required.
+In addition to the provided standard attributes, you can configure [Custom Attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes). These are attributes that are typically unique to your use case, such as a tenant ID or a user's display name. Custom attributes are identified by the `custom:` prefix:
 
-</Callout>
+```ts title="amplify/auth/resource.ts"
+import { defineAuth } from "@aws-amplify/backend"
+
+export const auth = defineAuth({
+  loginWith: {
+    // this configures a required "email" attribute
+    email: true,
+  },
+  userAttributes: {
+    // highlight-start
+    "custom:display_name": {
+      dataType: "String",
+      mutable: true,
+      maxLen: 16,
+      minLen: 1,
+    },
+    "custom:favorite_number": {
+      dataType: "Number",
+      mutable: true,
+      min: 1,
+      max: 100,
+    },
+    "custom:is_beta_user": {
+      dataType: "Boolean",
+      mutable: true,
+    },
+    "custom:started_free_trial": {
+      dataType: "DateTime",
+      mutable: true,
+    },
+    // highlight-end
+  },
+})
+```
+
+Unlike standard attributes, custom attributes cannot natively be required for sign-up, however can be codified to require some value by [validating user attributes upon sign-up with a pre sign-up trigger](/[platform]/build-a-backend/functions/user-attribute-validation/).
+
+Custom attributes can also be configured with specific data types. The following data types are supported:
```

Comment:
might be good to also list out max value range we can set or link to the doc that provides thus

```
- `String` : upto 2048 bytes
- `Number` : 2^1023
- `Boolean`
- `DateTime`
```

Reply:
ya i think it would be good, since it might make it easier to find